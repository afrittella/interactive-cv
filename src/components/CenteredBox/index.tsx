import React from 'react';
import { IStyledBase } from 'helpers/definitions';
import styled from 'styled-components';

const CenteredBox: React.FunctionComponent<IStyledBase> = ({
    className,
    children,
}) => {
    return (
        <div className={className}>
            <div className={`${className}__box`}>{children}</div>
        </div>
    );
};

export default styled(CenteredBox)`
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100vw;
    height: 100vh;
    background-color: ${props => props.theme.colorDark};
    color: ${props => props.theme.colorMedium};

    &__box {
        text-align: center;
        max-width: 80vw;
        padding: ${props => props.theme.sSize} ${props => props.theme.mSize};

        h3 {
            text-transform: uppercase;
        }
    }
`;
