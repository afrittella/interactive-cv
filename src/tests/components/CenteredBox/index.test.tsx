import React from 'react';
import { render, getByText } from '@testing-library/react';
import CenteredBox from 'components/CenteredBox';

test('renders a CenteredBox', () => {
    const component = <p>Test</p>;
    const { container } = render(<CenteredBox>{component}</CenteredBox>);

    const expectedComponent = getByText(container, 'Test');

    expect(container.firstChild).toMatchSnapshot();
    expect(expectedComponent).toBeInTheDocument();
});
