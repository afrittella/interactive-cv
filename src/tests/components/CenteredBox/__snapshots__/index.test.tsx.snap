// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a CenteredBox 1`] = `
<div
  class="sc-AxjAm jCwkpD"
>
  <div
    class="sc-AxjAm jCwkpD__box"
  >
    <p>
      Test
    </p>
  </div>
</div>
`;
